fuzz/target.c

```diff
@@ -260,6 +261,49 @@ shuffle_minimise(const char *pattern)
 	return EXIT_SUCCESS;
 }
 
+static int
+fuzz_print_vm(FILE *f, const char *pattern, bool det, bool min)
+{
+	assert(pattern != NULL);
+
+	struct re_err err;
+	struct fsm *fsm;
+	const size_t length = strlen(pattern);
+
+	struct scanner s = {
+		.str    = (const uint8_t *)pattern,
+		.size   = length,
+		.offset = 0
+	};
+
+	fsm = re_comp(RE_PCRE, scanner_next, &s, &opt, RE_MULTI, &err);
+
+	if (fsm == NULL) {
+		/* ignore invalid regexp syntax, etc. */
+		return EXIT_SUCCESS;
+	}
+
+	if (det) {
+		if (!fsm_determinise(fsm)) {
+			assert(!"det failure");
+			return EXIT_FAILURE;
+		}
+
+		if (min) {
+			if (!fsm_minimise(fsm)) {
+				assert(!"min failure");
+				return EXIT_FAILURE;
+			}
+		}
+	}
+
+	/* see if this triggers any asserts */
+	fsm_print_vmc(f, fsm);
```

Comment:
I think we should call all fsm_print interfaces here. This fuzzer isn't really specific to the VM-based outputs, either. Let's run everything we have.

Reply:
Doing so immediately revealed that several of them assert or `exit(EXIT_FAILURE)` if called with an unsupported IO mode. Should they just return instead?

Reply:
Okay. I can commit calling all of them that but not my local changes to make them not exit/assert, then.